src/abstract-interpretation/processor.ts

```diff
@@ -1,7 +1,157 @@
-import { DataflowInformation } from '../dataflow/internal/info'
-import { NormalizedAst } from '../r-bridge'
+import {DataflowInformation} from '../dataflow/internal/info'
+import {NodeId, NormalizedAst, ParentInformation, RBinaryOp, RType} from '../r-bridge'
+import {CfgVertexType, extractCFG} from '../util/cfg/cfg'
+import {visitCfg} from '../util/cfg/visitor'
+import {guard} from '../util/assert'
+import {DataflowGraphVertexInfo, EdgeType, OutgoingEdges} from '../dataflow'
+
+const constraintMap = new Map<NodeId, Constraints>()
+
+class Stack<ElementType> {
+	private backingStore: ElementType[] = []
+
+	size(): number { return this.backingStore.length }
+	peek(): ElementType | undefined { return this.backingStore[this.size() - 1] }
+	pop(): ElementType | undefined { return this.backingStore.pop() }
+	push(item: ElementType): ElementType {
+		this.backingStore.push(item)
+		return item
+	}
+}
+
+interface IHandler<ValueType> {
+	name:  string,
+	enter: () => void
+	exit:  () => ValueType
+	next:  (value: ValueType) => void
+}
+
+type Intervals = Set<{min: number, minInclusive: boolean, max: number, maxInclusive: boolean}>
+
+interface Constraints {
+	node:      NodeId,
+	intervals: Intervals,
+	debugMsg:  string
+}
+
+function intervalFromNumber(n: number): Intervals {
+	return new Set([{min: n, minInclusive: true, max: n, maxInclusive: true}])
+}
+
+function getIntervalsOfDfgChild(node: NodeId, dfg: DataflowInformation): Intervals {
```

Comment:
Child appears to be a little vague. Maybe `leaf` is a better fit? Or just narrow it down to symbols atm.